vendor/engines/bulk_email/app/services/bulk_email/bulk_email_searcher.rb

```diff
@@ -7,15 +7,22 @@ class BulkEmailSearcher
     attr_reader :order_details, :search_fields
 
     DEFAULT_SORT = [:last_name, :first_name].freeze
-    SEARCH_TYPES = [:customers, :account_owners, :customers_and_account_owners, :authorized_users].freeze
+    USER_TYPES = %i(customers authorized_users account_owners).freeze
 
     def initialize(search_fields)
       @search_fields = search_fields
     end
 
+    def user_types
+      @user_types ||=
+        USER_TYPES & (search_fields[:bulk_email].try(:[], :user_types) || []).map(&:to_sym)
```

Comment:
I feel like the `try`/`||` pattern is not always as clear as it could be. Are we ever using anything in `search_fields` that's not `:bulk_email`? It might be easier to deal with this if we made `@search_fields = Array(search_fields[:bulk_email])` in the contructor. 

It might also be a little clearer here to introduce an intermediate variable for the user inputs.